ethereum/dex/trades/insert_1inch.sql

```diff
@@ -463,80 +574,204 @@ RETURN r;
-    AND block_time <= '2020-01-01'
-    AND project = '1inch'
-);
+-- WHERE NOT EXISTS (
+--     SELECT *
+--     FROM dex.trades
+--     WHERE block_time > '2019-01-01'
+--     AND block_time <= '2020-01-01'
+--     AND project = '1inch'
+-- )
+;
 
 -- fill 2020
 SELECT dex.insert_1inch(
     '2020-01-01',
-    '2021-01-01',
+    '2020-07-01',
     (SELECT max(number) FROM ethereum.blocks WHERE time < '2020-01-01'),
+    (SELECT max(number) FROM ethereum.blocks WHERE time <= '2020-07-01')
+)
+-- WHERE NOT EXISTS (
+--     SELECT *
+--     FROM dex.trades
+--     WHERE block_time > '2020-01-01'
+--     AND block_time <= '2021-01-01'
+--     AND project = '1inch'
+-- )
+;
+
+-- fill 2020
+SELECT dex.insert_1inch(
+    '2020-07-01',
+    '2021-01-01',
+    (SELECT max(number) FROM ethereum.blocks WHERE time < '2020-07-01'),
     (SELECT max(number) FROM ethereum.blocks WHERE time <= '2021-01-01')
 )
-WHERE NOT EXISTS (
-    SELECT *
-    FROM dex.trades
-    WHERE block_time > '2020-01-01'
-    AND block_time <= '2021-01-01'
-    AND project = '1inch'
-);
+-- WHERE NOT EXISTS (
+--     SELECT *
+--     FROM dex.trades
+--     WHERE block_time > '2020-01-01'
+--     AND block_time <= '2021-01-01'
+--     AND project = '1inch'
+-- )
+;
 
 -- fill 2021
 SELECT dex.insert_1inch(
     '2021-01-01',
-    now(),
+    '2021-07-01',
     (SELECT max(number) FROM ethereum.blocks WHERE time < '2021-01-01'),
+    (SELECT MAX(number) FROM ethereum.blocks where time <= '2021-07-01')
+)
+-- WHERE NOT EXISTS (
+--     SELECT *
+--     FROM dex.trades
+--     WHERE block_time > '2021-01-01'
+--     AND block_time <= now() - interval '20 minutes'
+--     AND project = '1inch'
+-- )
+;
+
+-- fill 2021
+SELECT dex.insert_1inch(
+    '2021-07-01',
+    '2022-01-01',
+    (SELECT max(number) FROM ethereum.blocks WHERE time < '2021-07-01'),
+    (SELECT MAX(number) FROM ethereum.blocks where time <= '2022-01-01')
+)
+-- WHERE NOT EXISTS (
+--     SELECT *
+--     FROM dex.trades
+--     WHERE block_time > '2021-01-01'
+--     AND block_time <= now() - interval '20 minutes'
+--     AND project = '1inch'
+-- )
+;
+
+-- fill 2022
+SELECT dex.insert_1inch(
+    '2022-01-01',
+    '2022-07-01',
+    (SELECT max(number) FROM ethereum.blocks WHERE time < '2022-01-01'),
+    (SELECT MAX(number) FROM ethereum.blocks where time <= '2022-07-01')
+)
+-- WHERE NOT EXISTS (
+--     SELECT *
+--     FROM dex.trades
+--     WHERE block_time > '2021-01-01'
+--     AND block_time <= now() - interval '20 minutes'
+--     AND project = '1inch'
+-- )
+;
+
+-- fill 2022
+SELECT dex.insert_1inch(
+    '2022-07-01',
+    now(),
+    (SELECT max(number) FROM ethereum.blocks WHERE time < '2022-07-01'),
     (SELECT MAX(number) FROM ethereum.blocks where time < now() - interval '20 minutes')
 )
-WHERE NOT EXISTS (
-    SELECT *
-    FROM dex.trades
-    WHERE block_time > '2021-01-01'
-    AND block_time <= now() - interval '20 minutes'
-    AND project = '1inch'
-);
+-- WHERE NOT EXISTS (
```

Comment:
nit, I think it's a good idea to leave a comment as to why something is commented out.

Reply:
fair, added a few comments. also postgres seems to disagree with this subquery even when data does exist.